card/mod/carrierwave/set/abstract/attachment/storage_type.rb

```diff
@@ -0,0 +1,195 @@
+attr_writer :bucket, :storage_type
+
+event :storage_type_change, :store,
+      on: :update, when: proc { |c| c.storage_type_changed? } do
+  return if storage_type.in? [:web, :coded]
+  return if @new_storage_type.in? [:web, :coded]
+
+  case storage_type
+  when :cloud
+    if @new_storage_type == :cloud
+      move_from_cloud_to_cloud
+    else
+      move_from_cloud_to_local
+    end
+  when :protected, :unprotected
+    if @new_storage_type == :cloud
+      move_from_local_to_cloud
+    end
+  end
+  @storage_type = @new_storage_type
+  #attachment.url
+  #binding.pry
+  # @bucket = @new_bucket
+  # @storage_type = @new_storage_type
+  #
+  # write_identifier
+end
+
+event :create_public_link, :integrate, on: :save,
+                                       when: proc { |c| c.unprotected? } do
+  return if File.exist? public_path
+  FileUtils.mkdir_p File.dirname(public_path)
+  File.symlink attachment.path, public_path
+end
+
+event :remove_public_link, on: :update,
+                           after: :storage_type_change,
+                           when: proc { |c| !c.unprotected? } do
+  return unless File.exist? public_path
+  FileUtils.rm public_path
+end
+
+def cloud?
+  storage_type == :cloud
+end
+
+def web?
+  storage_type == :web
+end
+
+def unprotected?
+  storage_type == :unprotected
+end
+
+def protected?
+  storage_type == :protected
+end
+
+def coded?
+  @mod
+end
+
+def load_from_mod
+  @mod
+end
+
+def load_from_mod= value
+  @mod = value
+  write_identifier
+  @store_in_mod = true if value
+end
+
+def mod= value
+  @mod = value
+  write_identifier
+  @store_in_mod = true if value
+end
+
+def mod_file?
+  return @mod if @store_in_mod
+  # when db_content was changed assume that it's no longer a mod file
+  return if db_content_changed? || !content.present?
+  case content
+  when %r{^:[^/]+/([^.]+)} then Regexp.last_match(1) # current mod_file format
+  when /^\~/               then false  # current id file format
+  else
+    if (lines = content.split("\n")) && (lines.size == 4)
+      # old format, still used in card_changes.
+      lines.last
+    end
+  end
```

Comment:
will do that when I replace the old mod file stuff with "storage_type = :coded" api